// Copyright 2023 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//    https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";

package guideline;

// option cc_api_version = 2;
// option java_api_version = 2;
option java_multiple_files = true;
option java_package = "com.google.research.guideline.proto";
option optimize_for = LITE_RUNTIME;

// Results of line detection.
message LineDetection {
  // The detected line segments.
  repeated LineDetectionSegment segment = 1;

  // Relative position of the line in the frame [-1, 1].
  // This will be unset if no line was detected in the frame.
  optional float x_position = 2;

  // The detected curve angle, if a curve was detected.
  optional float curve_angle_degrees = 3;
}

message LineDetectionSegment {
  optional LineDetectionPoint bottom = 1;
  optional LineDetectionPoint top = 2;
}

message LineDetectionPoint {
  optional float x = 1;
  optional float y = 2;
}
